articles/tutorials/building_2d_games/04_game_library/index.md

```diff
@@ -0,0 +1,126 @@
+---
+
+Throughout this tutorial series, we will be creating reusable classes that add extended behavior on top of what MonoGame offers natively.  Instead of adding these classes directly to our game project, we can create a separate class library project to contain them.  This approach offers several advantages, including:
+
+1.  **Reusability**: The classes can be easily reused in future game projects by simply adding a reference to the library.
+2.  **Organization**: Keeps game-specific code separate from reusable library code.
+3.  **Maintainability**: Changes to the library can benefit all games that use it.
+4.  **Testing**: The library code can be tested independently of any specific game.
+
+A class library is a project type that compiles into a Dynamic Link Library (DLL) instead of an executable.  It contains reusable code that can be referenced by other projects, making it perfect for sharing common functionality across multiple games.  MonoGame offers the *MonoGame Game Library* project template that can be used to create a class library.
+
+> [!NOTE]
+> By using the **MonoGame Game Library** template instead of a standard C# class library template, we get the correct MonoGame framework references and configuration automatically set up.  This saves time and ensures compatibility with MonoGame projects.
+
+## Create the Class Library
+Let's use the MonoGame Game Library project template to add a new class library project that will contain our reusable code.
+
+### Visual Studio Code
+To add the class library using the MonoGame Game Library project template in Visual Studio Code, perform the following:
+
+1. Open the *Command Palette* (View > Command Palette).
+2. Enter ".NET New Project" and select that command.
+3. Enter "MonoGame Game Library" and select it as the template to use.
+4. Name the project "MonoGameLibrary".
+5. When prompted for a location, use the default option which will put the new project in a folder next to your game project.
+6. Select "Create Project".
+
+### Visual Studio 2022
+To add the class library using the MonoGame Game Library project template in Visual Studio 2022, perform the following:
+
+1. Right-click the *MonoGameSnake* solution in the Solution Explorer panel.
+2. Choose Add > New Project from the context menu.
+3. Enter "MonoGame Game Library" in the search box, select that template, then click Next.
+4. Name the project "MonoGameLibrary".
+5. The location by default will put the new project in a folder next to your game project, you do not need to adjust this.
+6. Click "Create".
+
+### dotnet CLI
+To add the class library using the MonoGame Game Library project template with the dotnet CLI, perform the following:
+
+1. Open a new Command Prompt or Terminal window in the same directory as the *MonoGameSnake.sln* solution file.
+2. Enter the command `dotnet new mglib -n MonoGameLibrary`.  This will create the project and place it in a folder next to your game project.
```

Comment:
super nitpicky, but all of the other bulleted points are single sentences. semicolon instead of period and second sentence here to be consistent?

Reply:
Wasn't sure about breaking it with semi colons.   I reworked the sentences to be single sentence structures instead

```
1. Open a new Command Prompt or Terminal window in the same directory as the *MonoGameSnake.sln* solution file.
2. Enter the command `dotnet new mglib -n MonoGameLibrary` to create the project, placing it in a folder next to your game project.
3. Enter the command `dotnet sln add ./MonoGameLibrary/` to add the newly created class library project to the *MonoGameSnake.sln* solution file.
```

Reply:
that's clean too. lgtm